addons/bus/tests/test_websocket_caryall.py

```diff
@@ -185,6 +186,18 @@ def test_trigger_notification(self):
         self.assertEqual(notifications[0]['message']['type'], 'notif_type')
         self.assertEqual(notifications[0]['message']['payload'], 'another_message')
 
+    def test_trigger_notification_invalid_language(self):
+        websocket = self.websocket_connect()
+        self.session.context['lang'] = 'xk_XK'  # this should not crash the notif. dispatching
+        http.root.session_store.save(self.session)
+        self.subscribe(websocket, ['my_channel'], self.env['bus.bus']._bus_last_id())
+        self.env['bus.bus']._sendone('my_channel', 'notif_type', 'message')
+        self.trigger_notification_dispatching(["my_channel"])
+        notifications = json.loads(websocket.recv())
+        self.assertEqual(1, len(notifications))
+        self.assertEqual(notifications[0]['message']['type'], 'notif_type')
+        self.assertEqual(notifications[0]['message']['payload'], 'message')
+
```

Comment:
This is not a good test to name and craft an invalid locale `xk_XK` that doesn't exist. The test looks extremely artificial and not representative on something that we should officially support.

`fr_FR` is definitely a valid locale. It just so happens that it is not installed. The test should be about the locale not being installed, not "invalid". The nuance is important, because "invalid" can mean anything and we definitely don't want to make millions of tests to check many (but barely a small amount) of "invalid" format.

I don't mind that internally test crafted an uninstalled locale with a fake one, just because it might be executed with all locale installed. But make sure to name the test with the appropriate use-case of testing / simulating how it handle an uninstalled locale.

Reply:
@alexkuhn The point was that the socket is open by websitor visitor (guest), so the session lang will be filled by whatever the browser send as preferred language and that is valid for babel's `parse_locale()`[^1], so `Accept-Language: xk-XK` will result in having session context as `{'lang': 'xk_XK'}`; and there are many more languages that what we currently have in `res.lang` (either installed or not)

But the remark is valid, I've updated the test to use a real language code: `fr_LU`

[^1]: https://github.com/odoo/odoo/blob/d7cfef9c51461a595f3f46c7b91e5c56241d4af3/odoo/http.py#L1449

Reply:
Yes, I get that in theory client-side could forge any language like `xk-XK`, but in practice this is not a valid one and this goes outside of an expected value that we must support.

Again I don't mind being very defensive in our code, but the remark was more about test maintenance: each test is more LOCs to maintain, so it's crucial to make them as maintainable as possible. A major contribution to that is to name and write them in a way that is very relevant functionally.

I'd rename `invalid` to `unsupported` in the test name, just so that the name of test is already clear enough. "invalid" is too broad and forces us to read the test, which is a waste of time and effort when this could all be prevented with a better named test.